scripts/dev_scripts/copyright-checker.sh

```diff
@@ -29,25 +30,46 @@ done
 
 if [ ${#missing_copyright_files[@]} -ne 0 ]; then
     for f in "${missing_copyright_files[@]}"; do
+
+        # Don't allow this script to run on itself.
+        if [[ $0 == $f ]];then
+            echo "Cannot run the $0 on itself. Please fix the headers in this file manually."
+            exit 1
+        fi
+        missing_license_note=$(grep -i "$license_note" "$f")
         startyear=$(git log --format=%ad --date=format:%Y "$f" | tail -1)
         if [[ -z "${startyear// }" ]]; then
             startyear=$currentyear
         fi
         if [[ $f =~ .*\.(js$|java$|go$|dl$) ]]; then
             expected="\/\* Copyright \(c\) $startyear - $currentyear, Oracle and\/or its affiliates\. All rights reserved\. \*\/"
-            expected="$expected\n\/\* Licensed under the Universal Permissive License v 1.0 as shown at https:\/\/oss\.oracle\.com\/licenses\/upl\/\. \*\/"
+            if [ ${#missing_license_note} -eq 0 ]; then
+                expected="$expected\n\/\* $license_note \*\/"
+            fi
         elif [[ $f =~ .*\.(py$|tf$|sh$|yaml$) ]] || [[ "${f##*/}" = "Dockerfile" ]]; then
             expected="# Copyright \(c\) $startyear - $currentyear, Oracle and\/or its affiliates\. All rights reserved\."
-            expected="$expected\n# Licensed under the Universal Permissive License v 1.0 as shown at https:\/\/oss\.oracle\.com\/licenses\/upl\/\."
-
+            if [ ${#missing_license_note} -eq 0 ]; then
+                expected="$expected\n# $license_note"
+            fi
         fi
 
-        if ! grep -i -e "Copyright (c) .* Oracle and/or its affiliates. All rights reserved" "$f" 1>/dev/null;then
-            echo "Copyright header missing for $f"
-            sed -i "1s/^/$expected\n\n/" "$f"
+        # Find the first matching copyright line.
+        line_number=$(grep -m 1 -n -i -e "Copyright (c) .* Oracle and/or its affiliates. All rights reserved" "$f" | cut -d : -f 1)
```

Comment:
Because we only match this line but not the `license_note` line, does it means that if the `lincense_note` line if missing won't be added if this line is already existed?. For example with this file:
```
#!/usr/bin/env bash

# Copyright (c) 2022 - 2023, Oracle and/or its affiliates. All rights reserved.

# other stuffs
```
The `copyright-checker.sh` script won't raise any issue.

Reply:
The script currently mainly checks if the copyright is missing or needs an update. To check if the license is missing only we would need an extra check at the beginning of the script, but that was not the focus of this fix. It should be fine for now as anytime a new file is added the license will be added too, and right now all the files have licenses. I will add an issue to specifically check for the missing license.